galaxy_importer/loaders/content.py

```diff
@@ -175,6 +175,22 @@ def _make_path_name(rel_path, name):
 
 
 class RoleLoader(ContentLoader):
+    def __init__(
+        self,
+        content_type,
+        rel_path,
+        root,
+        doc_strings=None,
+        cfg=None,
+        logger=None,
+        legacy=False,
+    ):
+        super().__init__(content_type, rel_path, root, doc_strings, cfg, logger)
+        self.legacy = legacy
+
+        if self.legacy:
+            self.path_name = self.name
```

Comment:
another option... you could add a `**kwargs` to the parent class init signature along with `self.kwargs = kwargs`, then the child class can have additional named params, and you could then check for self.kwargs.get("legacy_role") or whatever in the child `__init__()` or child `load()`

either approach may be fine, but seeing the init params duplicated (and therefore needing to be maintained in 2 places) made me think of that